### PR TITLE
Adding mark multiple posts as read, and other API changes.

### PR DESCRIPTION
**Pull request by dessalines**

### CODE REVIEW

File: src/types/CommentAggregates.ts [hunk 1]
```diff
@@ -9,6 +9,4 @@ export interface CommentAggregates {
   downvotes: /* integer */ number;
   published: string;
   child_count: number;
-  hot_rank: number;
-  controversy_rank: number;
```
Comment (SleeplessOne1917) [inline src/types/CommentAggregates.ts:13]: Is there supposed to be a controversial sort type to go along with this controversy rank?

**Comment by SleeplessOne1917** [comment]
1. Should there be new types for importing/exporting settings?
2. Add methods for user settings import/export
3. Should we get the animated images setting in this PR as well?

File: src/types/CommentAggregates.ts (hunk 1 above)
Comment (dessalines) [reply src/types/CommentAggregates.ts:13]: Yep, its in `SortType` and `CommentSortType`.

**Comment by dessalines** [comment]
> 1 & 2

Good call, I'll add those.

> 3 

That looks merged now, so I'll update it.

**Comment by dessalines** [comment]
@Nutomic Since the `UserSettingsBackup` struct is not in the `api_common` crate like the other API responses / requests, and it doesn't use ts-rs, its not properly creating the type, so front ends can't use it.

I tried for a bit to fix it, but unfortunately the structs use `ObjectId<ApubCommunity...` which isn't accessible from `api_common` (and those should be vectors of `DbUrl` s anyway).

**Comment by dessalines** [comment]
Merging so I can get a lemmy-ui RC out.

**Pull request merged**